docs/design-guides/architecture-overview.md

```diff
@@ -0,0 +1,49 @@
+- The Azure Active Directory is configured to support single tenant use case. 
+> Note: You can verify the configuration through app registration used for user sign in flow for Azure Active Directory instance. Go to the specific app registration within Azure Active Directory through [Azure Portal](https://portal.azure.com/) and check the Authentication tab to verify the tenancy configuration.
+- There are different Active Directory instances configured for different environments. 
+> Note: The sample only supports identity mapping for single environment within Graph Open Extensions used for storage of identity mapping.
+
+## Components
+![Diagram](../images/ACS-Authentication-Server-Sample_Overview-Flow.png)
+
+As seen from the Overview Diagram, the key components of the solution are:
+1. [Secure Web Api backed by Azure Active Directory](./secured-web-api-design.md)
+2. [Identity Mapping leveraging Graph Open Extensions](./Identity-Mapping-Design_Graph-Open-Extensions.md)
+3. Azure Communication Identity service which generates ACS Identity and Tokens. The interaction with ACS Identity service happens within [ACS sdk](https://docs.microsoft.com/en-us/azure/communication-services/concepts/sdk-options#sdks) used within the solution. 
+> Note: The `api/token/teams` endpoint does not leverage #2, as the M365 Azure Active Directory Identity is mapped to user's Teams Identity within ACS Identity Service.
+
+### Motivation for leveraging Graph Open Extensions for Identity Mapping
+The primary motivation comes from the fact that the Azure Communication Identity for the user could be co-located with rest of the information for the Active Active Directory user. And as that would also prevent the usage of additional storage for storing ACS Identity mappings corresponding to Azure Active Directory user Id.
+
+## Limitations
+- The open extensions on Azure Active Directory can only be supported upto 2 per user instance as suggested [here](https://docs.microsoft.com/en-us/graph/extensibility-overview#open-extension-limits). Hence supporting multi tenancy could be a challenge as different extensions might be desired for different tenants.
+- There is Graph open extensions rate limit of 455 requests per 10 seconds as suggested [here](https://docs.microsoft.com/en-us/graph/throttling#open-and-schema-extensions-service-limits)
+- The sample solution leveraging Graph open extensions might not be suitable for the scenarios where the Azure Active Directory instance is used for 3rd party application sign in as with [delegated permissions granted over Graph Api](https://docs.microsoft.com/en-us/graph/auth/auth-concepts#delegated-and-application-permissions). The 3rd Party application in this case with right delegated permissions would be able to access the open extension data of the user.
```

Comment:
This is unclear to me 😕

Reply:
This was something as an additional point from PM of the Open Extensions. So this is to say that if a 3rd party application is granted permissions on active directory instance of Company Contoso through app registration which has access to e.g. `user.read` permissions of contoso's   Active Directory user, then the open extensions data such as ACS user Id would be visible to 3rd party as well.
Since in this case ACS identity is EUPI, then Contoso should not be sharing that info. However I need to confirm this with Haider once he is back to office tomorrow/this week.

In general within team we feel it should not be concern given the active directory has more sensitive info e.g. address and stuff like that. So the sensitivity level of the ACS user Id should set the right guidance here for the users. Currently  its vague and left to user's choice how they want to handle that if they are fine sharing it or not.